src/cpp/desktop/DesktopMainWindow.cpp

```diff
@@ -671,44 +682,43 @@ void MainWindow::onLoadFinished(bool ok)
    {
       if (ok)
       {
-         // if this was a successful load, we're done
-         loadTimer_->stop();
-         loadedSuccessfully_ = true;
+         // we've loaded successfully; reset the reload count
+         // and stop the reload timer if it happens to be running
+         reloadTimer_->stop();
+         s_reloadCount = 0;
       }
-      else if (loadedSuccessfully_)
+      else if (s_reloadCount < s_maxReloadTries)
       {
-         // if the load was purportedly not successful, even
-         // though a prior load was successful, then ignore
-         // that (assume that this was a spurious / incorrect
-         // signal from Qt)
-         LOG_DEBUG_MESSAGE(
-                  "Discarding onLoadFinished(false) signal as we have "
-                  "already received onLoadFinished(true)");
+         // we failed to load; however, this load failure might be
+         // an intermittent failure. try again a couple times after
+         // a brief wait. note that we'll re-receive the onLoadFinished()
+         // signal after the webview has finished reloading
+         s_reloadCount += 1;
+         reloadTimer_->start(100);
```

Comment:
It feels like the problem here is that for some period of time the R session is refusing to accept connections. With this change, we will give up after trying for a total of 500ms, which feels like it might not be long enough. I think 5 retries is plenty but we might want to space them out more. How would you feel about trying this every 200ms instead (so we ultimately have a 1s grace period)?

Reply:
That sounds good to me. I'll make that change.